### PR TITLE
config window can now shuffle mobbers

### PR DESCRIPTION
**Pull request by ThomasAwesome**
We wanted a way to shuffle the mobbers to mix things up. We figured others would want it to.

### CODE REVIEW

File: src/client/config/index.css [hunk 1]
```diff
@@ -74,6 +74,16 @@ input {
...
+.btnShuffle {
+  background-color: var(--button-background-color);
+  border: 0;
+  border-radius: 3px;
+  padding: 7px 10px;
+  color: var(--button-text-color);
+  margin-bottom: 5px;
+  cursor: pointer;
+}
+
```
Comment (allan-stewart) [inline src/client/config/index.css]: This whole section looks to be just a copy of the `.btn` style. Is there a reason to add this? Why not just use the existing style?

File: src/service/__specs__/mobbers.spec.js [hunk 2]
```diff
@@ -16,6 +16,27 @@ describe('state/mobbers', () => {
...
+      let areSameArray = true;
+      for(let i = 0; i < mobberCopy.length; i++)
+      {
+        if(mobberCopy[i] !== subject.getAll()[i])
+        {
+          areSameArray = false;
+          break;
+        }
+      }
+      expect(areSameArray).equal(false);
```
Comment (allan-stewart) [inline src/service/__specs__/mobbers.spec.js]: Feels like this check is more complicated than it needs to be. Why not do a deep equality comparison, or at least do a map & join then check for string equality? The for loop logic makes the test code less readable.

Comment (allan-stewart) [reply src/service/__specs__/mobbers.spec.js]: Also, the implementation is using Math.random(), so isn't there a chance that you'll get the same array order back? Would that cause transient failures in the test? If so, maybe we should shuffle multiple times and make sure that most of them don't end up the same?

Alternatively, you could abstract or hijack (I think that sinon can do this) the random number generation to make it deterministic for your test.

File: src/service/timer-state.js [hunk 3]
```diff
@@ -125,6 +125,12 @@ class TimerState {
     this.persist()
   }
 
+  shuffleMobbers() {
```
[the remark this answers is not in the document]
Comment (allan-stewart) [reply src/service/timer-state.js]: I think you can follow the patterns in `timer-state.spec.js`. You can definitely assert the persistance that way. You might be able to verify that `this.mobbers.shuffleMobbers()` was called, but if that's not straightforward then you can put a few mobbers in and make sure that after the shuffe the order is different.

File: src/client/config/index.css (hunk 1 above)
Comment (ThomasAwesome) [reply src/client/config/index.css]: We needed a margin in the bottom to give some space between the mobbers and the button

Comment (allan-stewart) [reply src/client/config/index.css]: In that case, you should probably use both classes and slim this css to just be what's different. Then you can just set `class="btn btnShuffle"` in the html

**Comment by allan-stewart** [comment]
After some discussion with Parker, we decided to re-write history so that the significant restructuring could be done in a more collaborative and focused way. Unfortunately, that impacts this PR because it was based on that restructuring. Let me know how I can help with this.

**Pull request closed without merging**